File: src/1313_decompress-run-length-encoded-list.py
class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        output = []

        for idx in range(1, len(nums), 2):
            freq, val = nums[idx - 1], nums[idx]
            output += freq * [val]

        return output

File: Elixir/1313_decompress-run-length-encoded-list.ex
defmodule Solution do
  @spec decompress_rl_elist(nums :: [integer]) :: [integer]
  def decompress_rl_elist(nums) do
    Enum.chunk_every(nums, 2) |> Enum.flat_map(fn [freq, val] -> List.duplicate(val, freq) end)
  end
end
